### PR TITLE
cli: refactor exit codes + add api & auth err code

### PR DESCRIPTION
**Pull request by danielnorberg**

### CODE REVIEW

**Comment by danielnorberg** [comment]
@honnix PTAL

File: styx-cli/src/main/java/com/spotify/styx/cli/CliMain.java [hunk 1]
```diff
@@ -117,7 +118,7 @@ public static void main(String... args) {
     try {
       run(CliContext.DEFAULT, args);
     } catch (CliExitException e) {
-      System.exit(e.code());
+      System.exit(e.code().code);
```
Comment (honnix) [inline styx-cli/src/main/java/com/spotify/styx/cli/CliMain.java]: code code. :D Maybe e.exitCode().code or e.code().value ?

**Comment by honnix** [comment]
LGTM apart from the nitpicking.

**Comment by honnix** [comment]
+1

**Pull request merged**